horovod/tensorflow/__init__.py

```diff
@@ -75,7 +77,10 @@ def allreduce(tensor, average=True, device_dense='', device_sparse='',
         with tf.device(device_dense):
             horovod_size = tf.cast(size(), dtype=tensor.dtype)
             tensor_compressed, ctx = compression.compress(tensor)
-            summed_tensor_compressed = _allreduce(tensor_compressed)
+            if (group_id is not None):
```

Comment:
What if we just set the default to -1?  Then we could simplify these four lines to:

```
summed_tensor_compressed = _allreduce(tensor_compressed, group_id)
```

Personally, I think that would make the code simpler to parse (fewer special cases to consider).

Reply:
Sounds good to me. Changed default to -1 and removed extra cases.